components/timezone/src/time_zone.rs

```diff
@@ -175,88 +141,100 @@ impl CustomTimeZone {
         zone_offset_calculator: &ZoneOffsetCalculator,
         local_datetime: &DateTime<Iso>,
-    ) -> &mut Self {
-        if let Some(offset) = self.offset {
-            if let Some(time_zone_id) = self.time_zone_id {
-                if let Some((std, dst)) = zone_offset_calculator
-                    .compute_offsets_from_time_zone(time_zone_id, local_datetime)
-                {
-                    if offset == std {
-                        self.zone_variant = Some(ZoneVariant::standard());
-                    } else if Some(offset) == dst {
-                        self.zone_variant = Some(ZoneVariant::daylight());
-                    }
-                }
-            }
+    ) -> FormattableTimeZone {
+        FormattableTimeZone {
+            timezone: self,
+            metazone_id: self.bcp47_id.and_then(|id| {
+                metazone_calculator.compute_metazone_from_time_zone(id, local_datetime)
+            }),
+            zone_variant: self.bcp47_id.and_then(|id| {
+                self.offset.and_then(|offset| {
+                    zone_offset_calculator
+                        .compute_offsets_from_time_zone(id, local_datetime)
+                        .and_then(|(std, dst)| {
+                            if offset == std {
+                                Some(ZoneVariant::standard())
+                            } else if Some(offset) == dst {
+                                Some(ZoneVariant::daylight())
+                            } else {
+                                None
+                            }
+                        })
+                })
+            }),
         }
-        self
     }
 }
 
 #[cfg(feature = "compiled_data")]
-impl core::str::FromStr for CustomTimeZone {
+impl core::str::FromStr for TimeZone {
     type Err = UnknownTimeZoneError;
 
     #[inline]
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         Self::try_from_str(s)
     }
 }
+
+/// [`TimeZone`] annotated with additional formatting information.
+///
+/// This is obtained from [`TimeZone::to_formattable_at`].
+#[derive(Debug, Copy, Clone)]
+pub struct FormattableTimeZone {
+    timezone: TimeZone,
+    metazone_id: Option<MetazoneId>,
+    zone_variant: Option<ZoneVariant>,
+}
+
+impl FormattableTimeZone {
+    /// Creates a new [`FormattableTimeZone`] for the UTC time zone.
+    pub fn utc() -> Self {
+        Self {
+            timezone: TimeZone::utc(),
+            metazone_id: Some(MetazoneId(tinystr::tinystr!(4, "utc"))),
+            zone_variant: Some(ZoneVariant::standard()),
+        }
+    }
+
+    /// The UTC offset.
+    pub fn offset(&self) -> Option<UtcOffset> {
+        self.timezone.offset
+    }
+
+    /// The BCP47 time-zone identifier
+    pub fn bcp47_id(&self) -> Option<TimeZoneBcp47Id> {
+        self.timezone.bcp47_id
+    }
+
+    /// The CLDR metazone identifier
+    pub fn metazone_id(&self) -> Option<MetazoneId> {
+        self.metazone_id
+    }
```

Comment:
Issue: I do still feel it is useful/necessary to give a way to make one of these from IDs without data.

If we keep `CustomTimeZone`, it gives that low-level power for anyone who needs it.

Reply:
There is already a way to make one of these IDs without data: `MetazoneId(tinystr!(_, "usce"))`. If we teach the datetime formatter to format this, our job is done. I *do not* want anyone to be able to construct the combination of `chzrh` and `usce`.

Reply:
> If we teach the datetime formatter to format this

Ok.